src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -540,6 +540,7 @@ public static object GetDefaultValue(Type type)
 
         /// <summary>
         /// Marks the specified entity for deletion. The entity will be deleted when SaveChanges is called.
+        /// <para>WARNING: This method will call <see cref="IDocumentSession.Delete"/> if called during OnBeforeStore listener!</para>
```

Comment:
What? I don't understand this comment

Reply:
Calling delete will call delete?

Reply:
If `Delete` (by entity) called during `OnBeforeStore` then `Delete` (by id) will be used instead
I added this warning because `Delete` (by id) don't call `OnBeforeDelete` listener

Reply:
Let's remove the comment and make the `Delete(id)` raise the `OnBeforeDelete` as well

Reply:
IIRC this is a deferred command and we do not raise events for those and OnBeforeDelete requires an entity

Reply:
We can pass null for the entity, no?
And there is a different between `Delete(id)` and `Advanced.Defer()`

Reply:
The Delete(entity) should trigger the event for sure, the Delete(id) should not, I will check tomorrow with @garayx and we will see what can be done about that